### ISSUE TITLE
OWA / New Outlook setSignatureAsync inserts at incorrect position with S/MIME encrypted emails

### ISSUE THREAD

**Issue by MrDaleQ** [OfficeDev/office-js#4267]
When replying to a received email, setSignatureAsync inserts a signature at the wrong location when used with an email encrypted with S/MIME in OWA and New Outlook  The behaviour is not experienced with Classic Outlook.

## Your Environment
OWA / New Outlook (all versions)
* Platform: OWA / PC
* Host: Outlook
* Outlook version number: 20240308003.10 

## Expected behavior
As Classic Outlook the signature should be inserted above the email being replied to.

## Steps to reproduce
1. Receive an email encrypted with S/MIME in OWA or New Outlook
2. Reply to the email
3. call setSignatureAsync to insert a signature
4. Signature is inserted below the original email body

## Context
This functionality correctly places the signature in Classic Outlook.

**Comment by shighosh-msft** [comment]
Hi @MrDaleQ, this is not a repro for us. Could you please share a repro video for this issue and a sample add-in that we can use to debug this? You can create a private GH repo with the artifacts uploaded and share with @exextoc. Thanks!

**Comment by ivanosevitch** [comment]
We can reproduce a similar behavior with:

- Platform: OWA / PC
- Host: Outlook
- Client version: 20240315003.14
- Microsoft Outlook Version 1.2024.319.300

Signature is inserted below the original email body and not above.

Will see if we can share a repro video or sample addin.

**Comment by MrDaleQ** [comment]
@shighosh-msft I am confirming that I am able to reproduce the issue with the sample add-in, but whilst this is added to Classic Outlook, it is taking a while to be deployed to New Outlook.

Once this completes and I can confirm I will share both

**Comment by MrDaleQ** [comment]
https://github.com/MrDaleQ/MSShare/issues/1 @exextoc 

Our original Outlook sample addin no longer functions with "new" Outlook. I am speaking to our Dev team about this issue.

**Comment by MrDaleQ** [comment]
@exextoc there is a gif of the manifested issue in the repo above.

**Comment by akagarwa-msft** [comment]
We are not able to repro the issue at our end. Kindly add the code in the repo shared.

**Comment by MrDaleQ** [comment]
@akagarwa-msft the app manifest can be downloaded from outlook.signature365.com/fontmanifest.xml

The add-in works correctly in OWA and Classic Outlook. In New Outlook there is no attempt to open the taskpane when the button is cliked from the "Apps" window, and event based activation does not trigger in New Outlook.

![image](https://github.com/OfficeDev/office-js/assets/140503143/c47f4b45-7c8c-454d-a649-0a7911c9b93f)


Sharing the the prod manifest would require you to sign up to a trial of our service to use.

**Comment by rajjha-msft** [comment]
Hey @MrDaleQ 

Can you share a repro video with us ? Keep the console log open when you repro, and share the same.

**Comment by MrDaleQ** [comment]
Hi @rajjha-msft,

Can i just confirm that this is for the SMIME issue as opposed to the secondary issue of New Outlook not opening the fonttester addin?

If so, there is a gif showing the behaviour at the repo listed above, https://github.com/MrDaleQ/MSShare/issues/1

exextoc has access to this, but I have also added your account. 

Can you please share a method to open the console log within New Outlook - we are currently unaware of a method to do this.

**Comment by MrDaleQ** [comment]
Please see console.gif in the repo linked above.

**Comment by MrDaleQ** [comment]
@exextoc @shighosh-msft 

We now have an example of this occurring with a standard email - I am confirming with the customer permission to share the .msg file on the github link

**Comment by MrDaleQ** [comment]
Latest upload in repo exhibits this behaviour and is not S/MIME encrypted. Replication:

Open Msg file
Reply / forward
SetSigAsync inserts the signature below the previous email.

**Comment by rkpathak** [comment]
Thanks for reporting this issue. We are able to reproduce with the mail provided. This issue has been put on our backlog. We unfortunately have no timelines to share at this point. 
 

Internal tracking id: Office: [269407]

**Comment by sami-romdhana** [comment]
We have the same behavior, trying to isolate it I could manage to replicate by having a single SharePoint URL within the email following the format https://tenant.sharepoint.com/:x:/s/sitename/xxxxxxxxxxxxxxxxxxxxxxxxxxx with a valid file.

@MrDaleQ Not sure if that's what you have experienced, if that's not the case we'll open a new ticket to avoid mixing up

**Comment by ajays-msft** [comment]
Internal tracking id: Office: [269407]

**Comment by MrDaleQ** [comment]
@sami-romdhana Thanks for the heads up - we have 2 instances of the issue - one is related to an included valid sharepoint URL also - the other does not have this in the email body though

**Comment by johnny-outreach** [comment]
Is there any update on this request?

**Comment by akagarwa-msft** [comment]
The rollout for this fix is now complete and the change is available in client version [20240520053] or greater.

**Comment by MrDaleQ** [comment]
We belive this issue is resolved but are waiting on confirmation from our customers.
